core/src/client/lwm2m_static.c

```diff
@@ -318,6 +318,93 @@ int AwaStaticClient_Process(AwaStaticClient * client)
     return result;
 }
 
+static AwaLwm2mResult AwaStaticClientDefaultHandler(AwaStaticClient * client, AwaOperation operation, AwaObjectID objectID, 
+                                                    AwaObjectInstanceID objectInstanceID, AwaResourceID resourceID, AwaResourceInstanceID resourceInstanceID, 
+                                                    void ** dataPointer, uint16_t * dataSize, bool * changed)
+{
+    AwaLwm2mResult result;
+
+    ObjectDefinition * objectDefinition = Definition_LookupObjectDefinition(Lwm2mCore_GetDefinitions(client->Context), objectID);
+    if (objectDefinition != NULL)
+    {
+        uint8_t * offset;
+        ResourceDefinition * resourceDefinition;
+
+        // check instance range
+        if ((objectInstanceID >= 0) && (objectInstanceID < objectDefinition->MaximumInstances))
+        {
+            switch (operation)
+            {
+            case AwaOperation_CreateObjectInstance:
+                result = AwaLwm2mResult_SuccessCreated;
+                break;
+
+            case AwaOperation_DeleteObjectInstance:
+                result = AwaLwm2mResult_SuccessDeleted;
+                break;
+
+            case AwaOperation_CreateResource:
+                result = AwaLwm2mResult_SuccessCreated;
+                break;
+
+            case AwaOperation_DeleteResource:
+                break;
+
+            case AwaOperation_Write:
+            
+                resourceDefinition = Definition_LookupResourceDefinitionFromObjectDefinition(objectDefinition, resourceID);
+                if (resourceDefinition != NULL)
+                {
+                // TODO: check dataSize vs definition->DataElementSize
+                // TODO: what do we do about storing the length.. for opaque etc...
```

Comment:
The "WithPointer" functions make the API nice and clean for primative types, i.e integers, floats, but where are we supposed to store the length for opaque.

When we register a resource we supply a single pointer to the data. which in most cases is ok, however for the opaque case we need to store data + length.

Now we could use a pointer to a AwaOpaque type, which contains data and length. but in this case data is a void pointer, where we really need something statically allocated.